src/MoBi.Presentation/Tasks/SimulationCommitTask.cs

```diff
@@ -0,0 +1,126 @@
+            addNewInitialConditionsFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateInitialConditionsFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedInitialConditions));
+
+         moBiMacroCommand.Add(lastModuleConfiguration.SelectedParameterValues == null ? 
+            addNewParameterValuesFromSimulationChanges(simulationWithChanges, templateModule) : 
+            updateParameterValuesFromSimulationChanges(simulationWithChanges, lastModuleConfiguration.SelectedParameterValues));
+
+         return moBiMacroCommand.Run(_context);
+      }
+
+      private ICommand updateParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, ParameterValuesBuildingBlock selectedParameterValues)
+      {
+         return new MoBiEmptyCommand();
+      }
+
+      private ICommand updateInitialConditionsFromSimulationChanges(IMoBiSimulation simulationWithChanges, InitialConditionsBuildingBlock selectedInitialConditions)
+      {
+         return new MoBiEmptyCommand();
+      }
+
+      private IMoBiCommand addNewParameterValuesFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var parameterValuesToAdd = changesFrom<Parameter>(simulationWithChanges).Select(x => createParameterValue(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<ParameterValuesBuildingBlock, ParameterValue>(simulationWithChanges.Name, lastModule.ParameterValuesCollection.AllNames(), lastModule, parameterValuesToAdd);
+      }
+
+      private IMoBiCommand addNewInitialConditionsFromSimulationChanges(IMoBiSimulation simulationWithChanges, Module lastModule)
+      {
+         var initialConditionsToAdd = changesFrom<MoleculeAmount>(simulationWithChanges).Select(x => createInitialCondition(x.quantity, x.objectPath)).ToList();
+         return createAddBuildingBlockCommand<InitialConditionsBuildingBlock, InitialCondition>(simulationWithChanges.Name, lastModule.InitialConditionsCollection.AllNames(), lastModule, initialConditionsToAdd);
+      }
+
+      /// <summary>
+      ///    Returns path and quantity tuple for all changes in the <paramref name="simulationWithChanges" /> of type
+      ///    <typeparamref name="TQuantity" />
+      /// </summary>
+      private IEnumerable<(ObjectPath objectPath, TQuantity quantity)> changesFrom<TQuantity>(IMoBiSimulation simulationWithChanges) where TQuantity : Quantity
+      {
+         var quantities = _entitiesInSimulationRetriever.EntitiesFrom<TQuantity>(simulationWithChanges);
+         return simulationWithChanges.OriginalQuantityValues.Where(x => quantities.Contains(x.Path)).Select(x => (objectPath: x.Path, quantity: quantities[x.Path]));
+      }
+
+      private IMoBiCommand createAddBuildingBlockCommand<TBuildingBlock, TPathAndValueEntity>(string buildingBlockName, IEnumerable<string> usedNames, Module moduleToAddTo, IReadOnlyList<TPathAndValueEntity> entitiesToAdd)
```

Comment:
can we retrieve  IEnumerable<string> usedNames from the module 
e.g. modules.All<TBuildingBlock>.AllName(). Just trying to reduce the number of parameters if possible

Reply:
If you have all building blocks except PV's and they are all named like the simulation, then you wouldn't be able to create the PV building block with the same name as all the other building blocks.

I think the expected behavior of adding the first PV or IC from a commit would be to have the same name as the simulation, which is allowed.

Reply:
```module.BuildingBlocks.OfType<TBuildingBlock>().AllNames();```
That should work